File: EntityLayer/Entities/AssetPositionHistory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EntityLayer.Entities
{
    public class AssetPositionHistory
    {
        public int Id { get; set; }
        public int AssetId { get; set; }
        public double X { get; set; }
        public double Y { get; set; }
        public DateTime DateTime { get; set; }
        public int FloorMapId { get; set; }

        public virtual Asset Asset { get; set; } = null!;
        public virtual FloorMap FloorMap { get; set; } = null!;
    }
}

File: Software/core/src/main/java/hr/foi/air/core/models/impl/AssetPositionHistory.kt
package hr.foi.air.core.models.impl

import hr.foi.air.core.models.IAssetPositionHistory

class AssetPositionHistory(
    override var id: Int,
    override val assetId: Int,
    override val x: Float,
    override val y: Float,
    override val dateTime: String,
    override val floorMapId: Int
) : IAssetPositionHistory{

}